### PR TITLE
Gitlab docs improved; gitlab webhook secret config standardization

### PR DESCRIPTION
**Pull request by s1moe2**
### **User description**
- improved GitlLab setup docs:
  - access token role
  - secrets and vars configuration
  - docker build arguments (used to reference github app instructions, which have a different target)
  
- introduced `webhook_secret` in the gitlab config section, to make it consistent with the other webhook secret configs from other sections
  - `shared_secret` is kept for backwards compatibility

___

### **PR Type**
Documentation, Enhancement

___

### **Description**
- Improved GitLab setup documentation with detailed steps for creating access tokens, setting up webhooks, and building Docker images.
- Introduced `webhook_secret` in the GitLab configuration to standardize with other webhook configurations, while maintaining backward compatibility with `shared_secret`.
- Updated the secrets template to include `webhook_secret` and clarified its usage.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>gitlab_webhook.py</strong><dd><code>Add support for GitLab `webhook_secret` configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/servers/gitlab_webhook.py

<li>Added support for <code>webhook_secret</code> in GitLab webhook configuration.<br> <li> Maintained backward compatibility with <code>shared_secret</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1307/files#diff-c3ea1ba460930c6e6e6a2c36a7dc729d79730a96e872d40ffb246f80bc7c0880">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>gitlab.md</strong><dd><code>Improve GitLab setup documentation and instructions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/docs/installation/gitlab.md

<li>Updated GitLab setup instructions for access token and webhook.<br> <li> Added detailed steps for Docker image building and deployment.<br> <li> Standardized secret configuration instructions.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1307/files#diff-568bccbe96a50a16f738f60a0c9f79e9a0c71a68bd4185281218e4aaafd24ed6">+24/-9</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>.secrets_template.toml</strong><dd><code>Update secrets template for GitLab webhook configuration</code>&nbsp; </dd></summary>
<hr>

pr_agent/settings/.secrets_template.toml

<li>Added <code>webhook_secret</code> to GitLab section.<br> <li> Clarified <code>shared_secret</code> as backward compatible.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1307/files#diff-f9cf1bb39fd1a7e126b4d06af0238b24125e183f540c20360a392ef695682783">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: docs/docs/installation/gitlab.md [hunk 1]
```diff
@@ -42,21 +42,36 @@ Note that if your base branches are not protected, don't set the variables as `p
 
 ## Run a GitLab webhook server
 
-1. From the GitLab workspace or group, create an access token. Enable the "api" scope only.
+1. From the GitLab workspace or group, create an access token with "Reporter" role and "api" scope.
```
Comment (mrT23) [inline docs/docs/installation/gitlab.md:45]: should at least a 'developer'

https://docs.gitlab.com/ee/user/permissions.html


image

File: pr_agent/servers/gitlab_webhook.py [hunk 2]
```diff
@@ -159,8 +159,8 @@ async def inner(data: dict):
             except Exception as e:
                 get_logger().error(f"Failed to validate secret {request_token}: {e}")
                 return JSONResponse(status_code=status.HTTP_401_UNAUTHORIZED, content=jsonable_encoder({"message": "unauthorized"}))
-        elif get_settings().get("GITLAB.SHARED_SECRET"):
-            secret = get_settings().get("GITLAB.SHARED_SECRET")
+        elif get_settings().get("GITLAB.SHARED_SECRET") or get_settings().get("GITLAB.WEBHOOK_SECRET"):
+            secret = get_settings().get("GITLAB.SHARED_SECRET") or get_settings().get("GITLAB.WEBHOOK_SECRET")
```
Comment (mrT23) [inline pr_agent/servers/gitlab_webhook.py:163]: remove this duplication.

configs shoud have a single definition

File: docs/docs/installation/gitlab.md [hunk 3]
```diff
@@ -42,21 +42,36 @@ Note that if your base branches are not protected, don't set the variables as `p
...
-    - Set deployment_type to 'gitlab' in [configuration.toml](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml)
+3. Clone this repository:
 
-5. Create a webhook in GitLab. Set the URL to ```http[s]://<PR_AGENT_HOSTNAME>/webhook```. Set the secret token to the generated secret from step 2.
-In the "Trigger" section, check the ‘comments’ and ‘merge request events’ boxes.
+```
+git clone https://github.com/Codium-ai/pr-agent.git
+```
+
+4. Prepare variables and secrets. Skip this setp if you plan on settings these as environment variables when running the agent:
```
Comment (mrT23) [inline docs/docs/installation/gitlab.md:59]: setp - > step

File: docs/docs/installation/gitlab.md [hunk 4]
```diff
@@ -42,21 +42,36 @@ Note that if your base branches are not protected, don't set the variables as `p
...
+    - In the [gitlab] section, set `personal_access_token` (with token from step 1) and `webhook_secret` (with secret from step 2)
+
+
+5. Build a Docker image for the app and optionally push it to a Docker repository. We'll use Dockerhub as an example:
+```
+docker build . -t gitlab_pr_agent --target gitlab_webhook -f docker/Dockerfile
+docker push codiumai/pr-agent:gitlab_webhook  # Push to your Docker repository
+```
+
+6. Create a webhook in GitLab. Set the URL to ```http[s]://<PR_AGENT_HOSTNAME>/webhook```, the secret token to the generated secret from step 2, andenable the triggers `push`, `comments` and `merge request events`.
```
Comment (mrT23) [inline docs/docs/installation/gitlab.md:74]: andenable - > and enable

File: docs/docs/installation/gitlab.md [hunk 5]
```diff
@@ -42,21 +42,36 @@ Note that if your base branches are not protected, don't set the variables as `p
...
 
-6. Test your installation by opening a merge request or commenting or a merge request using one of CodiumAI's commands.
+7. Test your installation by opening a merge request or commenting or a merge request using one of CodiumAI's commands.
```
Comment (mrT23) [inline docs/docs/installation/gitlab.md:76]: or commenting or a merge ->
or commenting on a merge

**Comment by mrT23** [comment]
@s1moe2 Thanks for the PR.
 i gave some notes on typos, but the main feedback is regarding the duplication of a config.
This should be removed.

It is a shared_secret, and the name is used also in other places.

**Comment by mrT23** [comment]
p.s. 
a tip to find typos:

image
https://chromewebstore.google.com/detail/qodo-merge-ai-powered-cod/ephlnjeghhogofkifjloamocljapahnl

File: docs/docs/installation/gitlab.md (hunk 1 above)
Comment (s1moe2) [reply docs/docs/installation/gitlab.md:45]: I see. But wiki access is only possible with Pro features, right? I can detail that.

File: pr_agent/servers/gitlab_webhook.py (hunk 2 above)
Comment (s1moe2) [reply pr_agent/servers/gitlab_webhook.py:163]: The decision is yours, ofc. 
The idea was not to duplicate but rather to guarantee backwards compatibility.
As for the var name, I only "shared_secret" used in a very specific context on the bitbucket webhook handler (when using a secret provider). The goal was to make things consistent between providers. Let me know if this is final then.

File: docs/docs/installation/gitlab.md (hunk 1 above)
Comment (mrT23) [reply docs/docs/installation/gitlab.md:45]: ok, i guess you are right

File: pr_agent/servers/gitlab_webhook.py (hunk 2 above)
Comment (mrT23) [reply pr_agent/servers/gitlab_webhook.py:163]: Definitely remove, and we can merge.

the name appears in other places

image

File: docs/docs/installation/gitlab.md (hunk 1 above)
Comment (s1moe2) [reply docs/docs/installation/gitlab.md]: adjusted

File: pr_agent/servers/gitlab_webhook.py (hunk 2 above)
Comment (s1moe2) [reply pr_agent/servers/gitlab_webhook.py]: removed

**Review by mrT23 — APPROVED** [review]

**Pull request merged**